src/Campaigns/Blocks/CampaignForm/resources/components/ModalForm/index.tsx

```diff
@@ -0,0 +1,149 @@
+import {useEffect, useState} from '@wordpress/element';
+import {__} from '@wordpress/i18n';
+import IframeResizer from 'iframe-resizer-react';
+import {Button, Dialog, Modal, ModalOverlay} from 'react-aria-components';
+import ModalCloseIcon from './ModalClose';
+import {Spinner} from '@wordpress/components';
+import './styles.scss';
+import '../../../../shared/components/EntitySelector/styles/index.scss';
+
+/**
+ * @unreleasaed
+ */
+type ModalFormProps = {
+    dataSrc: string;
+    embedId: string;
+    buttonText: string;
+    isFormRedirect: boolean;
+    formViewUrl: string;
+};
+
+/**
+ * @unreleasaed
+ */
+export default function ModalForm({dataSrc, embedId, buttonText, isFormRedirect, formViewUrl}: ModalFormProps) {
```

Comment:
@JoshuaHungDinh is this duplicated from the form block? if so, we should probably make this a shared component between the two of them

Reply:
@jonwaldstein we have the shared directory within the blocks but there is not an alias setup.
Would you prefer moving them into @givewp/campaigns/resources or create a new alias?
The benefit of the new alias would be the hooks & other components have nicer imports as well. We can move those too but some of them are pretty Campaign block specific.

Reply:
I think we can just move into the shared folder inside campaign blocks.  it's okay if we import that into the donation form one since that's the only block left there

Reply:
Got it. Resolved: 8f91473154991756b87c8213dc109ec6d490e07c

Reply:
@JoshuaHungDinh you should be able to use the existing alias of `@givewp/src` or `@givewp/campaigns` instead of those long imports

Reply:
Yes! Thanks: 5c06e2771c33453b67020af293c6ebc71941e20b

Reply:
Thanks, but I they don't look quite right

<img width="487" alt="Screenshot 2025-04-24 at 11 55 00 AM" src="https://github.com/user-attachments/assets/04e04f4d-2cf7-474f-9448-3539a953e077" />

Reply:
Yea this Is why I brought up the alias question.  `@givewp/campaigns` goes into `campaign/resources` not the Campaigns main directory & my ide doesn't recognize `@givewp/src` as an alias.
https://github.com/impress-org/givewp/blob/develop/tsconfig.json

Reply:
We have the alias in webpack but it looks like perhaps typescript is not recognizing it https://github.com/impress-org/givewp/blob/74b705bf35a7f3fe7d980ee2ab06bc3bec032d44/webpack.config.js#L136

When I converted everything to wordpress/scripts I found that if alias's live in webpack.config then they become discoverable which might negate the need for the typescript specific alias.  still have to look into that

Reply:
My ide had tricked me yea. Your right. fbe054b044e673e207f6cf28abb86eb4b1888216